### PR TITLE
Updated git data commit response with signature verification object

### LINKED ISSUE #1254: Get Signature Verification When Viewing Git Data Commit
API documentation: https://developer.github.com/v3/git/commits/#commit-signature-verification

An enhancement as part of the Commit Signing API is to surface the signature details in the commit.

It looks like these are the new properties:

```json
    "verification": {
      "verified": true,
      "reason": "valid",
      "signature": "-----BEGIN PGP MESSAGE-----\n...\n-----END PGP MESSAGE-----",
      "payload": "tree 6dcb09b5b57875f334f61aebed695e2e4193db5e\n..."
    }
```

These fields should be available on the `Commit` response object (perhaps in a class called `Verification` or something), and the `CommitsClient.Get` action should be updated to use the new `Accepts` header:

> application/vnd.github.cryptographer-preview+sha

I'd also like to see a new integration test added to verify this behaviour is working as expected.

#### Questions

We use `Commit` in many places, so I hope that this doesn't affect other places where we deserialize the payload...

### PR DESCRIPTION
**Pull request by Sarmad93**
Fixes #1254  
Hi @ryangribble @shiftkey  could you plz review this code :smiley: Regarding Integration test i am little confused where should i put this method, is it fine here in `CommitsClientTests` ? And what do you think of this integration test, right now this test is passing on repository which is of my test account. Is it fine here to hard code(string literals) repository parameter? Your feedback on this is much needed so that we can make improvement to this.

### CODE REVIEW

File: Octokit/Models/Response/Commit.cs [hunk 1]
```diff
@@ -1,4 +1,6 @@
-﻿using System.Collections.Generic;
+﻿using Octokit.Internal;
```
Comment (shiftkey) [inline Octokit/Models/Response/Commit.cs]: :lipstick: reorder these so `System.*` is before the rest...

File: Octokit/Models/Response/Verification.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
+﻿using Octokit.Internal;
```
Comment (shiftkey) [inline Octokit/Models/Response/Verification.cs]: :lipstick: ordering

Comment (Sarmad93) [inline Octokit/Models/Response/Verification.cs]: ok...

File: Octokit.Tests.Integration/Clients/CommitsClientTests.cs [hunk 3]
```diff
@@ -43,4 +43,21 @@ public class CommitsClientTests
...
+        var commit = await github.Git.Commit.Get("noonari", "source-repo-20160517084127190", "1965d149ce1151cf411300d15f8d890d9259bd21");
+
+        Assert.False(commit.Verification.Verified);
+        Assert.Equal(commit.Verification.Signature,
+ "-----BEGIN PGP SIGNATURE-----\nVersion: GnuPG v1\n\niQEcBAABAgAGBQJXYT2BAAoJEJyZ1vxIV0+N9ZwIAKlf3dk9n1q1mD5AT3Ahtj9o\nF4H25zsHynJk2lnH4YxVvDBEc/uMCXzX6orihZiSdA5UXE7tPyEEZddQdp8pxulX\ncIsFKcrfQqHJnTbT90z5PhAk94lyN9fFngzPW1tgZZVjp2YiiqgXduBWWm6EREOh\nS1Iu9wBqScQomhTXoksmNZyGTZ0LviSi0pkqRY64pQhKnpLlu1OFXaeDvhYocB+E\nY5URZsXodvIkBuzCkWCu8ra4eaXIIARkas4+jIvn0FIx9CzEVz0Zau/5Fk+BR+Te\n7a3/7JH7yuObPB0hqPSuFYyxtvPfxtayvhkGD3YkQqDAkWCpISGyVFzxrrC7z0Y=\n=kbih\n-----END PGP SIGNATURE-----");
+
+        Assert.Equal(commit.Verification.Payload, 
+            "tree c91c844f37974093a3f0a864755441b577e7663a\nparent 6eb645f6badd46de65700b4d7b6fcdb97684ce5a\nauthor noonari <SarmadSattar1@gmail.com> 1465990529 +0500\ncommitter noonari <SarmadSattar1@gmail.com> 1465990529 +0500\n\ngpg stuff\n");
+
+        Assert.Equal(commit.Verification.Reason.ToString(), "UnknownKey");
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/CommitsClientTests.cs]: :lipstick: why not compare both enums here?

File: Octokit.Tests.Integration/Clients/CommitsClientTests.cs [hunk 4]
```diff
@@ -43,4 +43,21 @@ public class CommitsClientTests
             Assert.NotNull(retrieved);
         }
     }
+
+    [IntegrationTest]
+    public async Task CanDeserializeVerificationObjectInResponse()
+    {
+        var github = Helper.GetAuthenticatedClient();
+
+        var commit = await github.Git.Commit.Get("noonari", "source-repo-20160517084127190", "1965d149ce1151cf411300d15f8d890d9259bd21");
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/CommitsClientTests.cs]: Would it be worth renaming this repository to make it's purpose more clear?

**Comment by shiftkey** [comment]
@Sarmad93 looking good, just a few little things to clean up. Love that integration test 😍

**Comment by Sarmad93** [comment]
:+1:  for review. Fixing this.

File: Octokit/Helpers/AcceptHeaders.cs [hunk 5]
```diff
@@ -27,7 +27,9 @@ public static class AcceptHeaders
         public const string MigrationsApiPreview = "application/vnd.github.wyandotte-preview+json";
 
         public const string ReactionsPreview = "application/vnd.github.squirrel-girl-preview";
-        
+
+        public const string SignatureVerification = "application/vnd.github.cryptographer-preview+sha";
```
Comment (ryangribble) [inline Octokit/Helpers/AcceptHeaders.cs]: Maybe name this `SignatureVerificationPreview` just for consistency with all the other preview headers?

File: Octokit/Models/Response/Verification.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+        [Parameter(Value = "malformed_signature")]
+        MalformedSignature,
+
+        [Parameter(Value = "inavlid")]
+        Invalid,
+
+        [Parameter(Value = "valid")]
+        Valid
+
```
Comment (ryangribble) [inline Octokit/Models/Response/Verification.cs]: Extra whitespace line here

File: Octokit/Models/Response/Verification.cs [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+                     "Verification: {0} Verified: {1} Reason: {2} Signature: {3} Payload",
+                     Verified,
+                     Reason.ToString(),
+                     Signature,
+                     Payload);
+            }
+        }
+    }
+
+    public enum Reason
```
Comment (ryangribble) [inline Octokit/Models/Response/Verification.cs]: Is `Reason` perhaps too generic here?

Comment (Sarmad93) [inline Octokit/Models/Response/Verification.cs]: right. fixing this. :+1:

**Comment by ryangribble** [comment]
I'll let @shiftkey hit the button since he reviewed this one, but it looks good to me @Sarmad93

**Comment by Sarmad93** [comment]
Right..thanks for the review. :smile:

File: Octokit/Models/Response/Verification.cs [hunk 8]
```diff
@@ -0,0 +1,90 @@
+﻿using System.Diagnostics;
+using System.Globalization;
+using Octokit.Internal;
+
+namespace Octokit.Models.Response
```
Comment (shiftkey) [inline Octokit/Models/Response/Verification.cs]: There's a coding convention here that I guess needs to be documented properly. Despite these files living in `Octokit/Models/Responses` the corresponding namespace for these is just `Octokit`. This means new `using` statements are unnecessary for consumers (and for this PR).

File: Octokit/Models/Response/Commit.cs [hunk 9]
```diff
@@ -2,6 +2,7 @@
 using System.Collections.ObjectModel;
 using System.Diagnostics;
 using System.Linq;
+using Octokit.Models.Response;
```
Comment (shiftkey) [inline Octokit/Models/Response/Commit.cs]: This shouldn't be necessary when the namespace is updated as per the other comment below...

**Comment by shiftkey** [comment]
@Sarmad93 @ryangribble thanks for waiting for me. Just that one comment and I think this is good to go!

File: Octokit.Tests.Integration/Clients/CommitsClientTests.cs [hunk 10]
```diff
@@ -1,5 +1,6 @@
 ﻿using System.Threading.Tasks;
 using Octokit;
+using Octokit.Models.Response;
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/CommitsClientTests.cs]: As mentioned elsewhere, this should be unnecessary after renaming the namespace...

**Comment by Sarmad93** [comment]
@shiftkey 
> thanks for waiting for me

 yeh no worries. i already knew your absence when i saw your tweet about your off for a week :wink:

**Comment by shiftkey** [comment]
@Sarmad93 looks good, thanks for this!

**Pull request merged**